tests/lib/rules/v-if-else-key.js

```diff
@@ -424,6 +508,69 @@ tester.run('v-if-else-key', rule, {
           line: 6
         }
       ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <template>
+          <div>
+            <ComponentA v-if="foo" />
+            <ComponentA v-else />
+
+            <ComponentA v-if="bar" />
+            <ComponentA v-else-if="baz" />
+            <ComponentA v-else />
+          </div>
+        </template>
+        <script>
+        export default {
+          components: {
+            ComponentA
+          }
+        }
+        </script>
+        `,
+      output: `
+        <template>
+          <div>
+            <ComponentA key="component-a-1" v-if="foo" />
+            <ComponentA key="component-a-2" v-else />
+
+            <ComponentA v-if="bar" />
```

Comment:
I believe this should also have a key, but we're currently limiting the rule to one `if` per family. This seems like a rare case, and I'm unsure how to address it. Should we address it? Any ideas or suggestions?

Reply:
I also think that this should be reported. I don't think it's that rare (especially with `<div>`s), so let's better handle it.

I think all consecutive elements with the same element name should go into the same conditional family, i.e. `v-if` should not always create a new conditional family.

Reply:
got it!
Logic refactor:
[3f4d687](https://github.com/vuejs/eslint-plugin-vue/pull/2348/commits/3f4d6877311b2be4185a4b89a680fd1d39cb2f24)

Updated test:
4cf8e87

Reply:
Nice!